warehouse/sanity-check/main.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/concourse/dutyfree/fetcher"
+	"github.com/concourse/dutyfree/resource"
+	"github.com/fatih/color"
+	"github.com/gobuffalo/packr/v2"
+	"gopkg.in/yaml.v2"
+)
+
+func printColor(colour *color.Color, strToPrint ...string) {
+	if _, err := colour.Println(strToPrint); err != nil {
+		fmt.Println(strToPrint)
+	}
+}
+
+func main() {
+
+	ftchr := fetcher.Fetcher{
+		Box: packr.New("resources", "../../resource-types"),
+	}
+
+	files, err := ftchr.GetAll()
+	if err != nil {
+		panic(err)
+	}
+	red := color.New(color.FgRed, color.Bold)
+	green := color.New(color.FgGreen, color.Bold)
+	yellow := color.New(color.FgYellow, color.Bold)
+
+	errors := []string{}
+	warning := []string{}
+
+	client := &http.Client{
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+
+	for _, fileBytes := range files {
```

Comment:
should we have tests for this? (at least eventually?)

jw 👀

Reply:
by this i mean the entire mini sanity-check go app

Reply:
I honestly didn't bother testing it since it is not production code, I was more using it as a checking script that was easier written in go rather than in bash. But yeah, we should eventually test it and also refactor it!